File: JavaScript/class/vipi.js
class Vipi extends Composant {
  constructor(position_x, position_y) {
    super(position_x, position_y, 200, 200, "../../img/totem.png");
    this.vie = 121;
    this.hydratation = 121;
    this.energie = 121;
    this.ch2019 = true;
  }

  getVie() {
    return this.vie;
  }

  setVie(nouvelle_vie) {
    this.vie = nouvelle_vie;
  }

  getHydratation() {
    return this.hydratation;
  }

  setHydratation(nouvelle_hydratation) {
    this.hydratation = nouvelle_hydratation;
  }

  getEnergie() {
    return this.energie;
  }

  setEnergie(nouvelle_energie) {
    this.energie = nouvelle_energie;
  }

  getCh2019() {
    return this.ch2019;
  }

  changeCh2019() {
    if(this.ch2019 == false) {
      this.ch2019 = true;
    } else {
      this.ch2019 = false;
    }
  }
}

File: Vipi_javascript_de_merde/src/Class/Vipi.java
package Class;

import java.io.IOException;

public class Vipi extends Composant {
	private Integer vie = 121;
	private Integer hydratation = 121;
	private Integer energie = 121;
	private boolean ch2019 = true;
	
	public Vipi() throws IOException {
		super(0, 0, 200, 200, "../images/totem.png");
	}
	
	public Integer getVie() {
		return vie;
	}
	
	public void setVie(Integer vie) {
		this.vie = vie;
	}
	
	public Integer getHydratation() {
		return hydratation;
	}
	
	public void setHydratation(Integer hydratation) {
		this.hydratation = hydratation;
	}
	
	public Integer getEnergie() {
		return energie;
	}
	
	public void setEnergie(Integer energie) {
		this.energie = energie;
	}
	
	public boolean isCh2019() {
		return ch2019;
	}
	
	public void changeCh2019() {
		if(this.ch2019 == true) {
			this.ch2019 = false;
		} else {
			this.ch2019 = true;
		}
	}
}
